File: School/Ngoac.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <stack>
using namespace std;

string st;

void xuly(){
	stack<int> s;
	for(int i=0; i<st.size(); i++)
		if(st[i]==')'){
			cout<< s.top()+1<< " "<< i+1<< endl;
			s.pop();
		}else s.push(i);
}

int main(){
	freopen("Ngoac.inp","r",stdin);
	freopen("Ngoac.out","w",stdout);
	cin>> st;
	xuly();
	return 0;
}

File: School/Ngoac.py
inp = open("Ngoac.inp", "r")
# out = open("Ngoac.out","w")

st = inp.read()
stack = []

for i in range(len(st)):
    if st[i] == ")":
        print(stack.pop() + 1, i + 1)
    else:
        stack.append(i)
